packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -132,6 +136,92 @@ export class SamplerOperations {
             .catch(/* do nothing */);
     }
 
+    public async getTokenDecimalsAsync(tokens: Address[]): Promise<number[]> {
+        return (await this.rpcSamplerClient.getTokensAsync(tokens)).map(t => t.decimals);
+    }
+
+    public async getSellQuotesAsync(
+        sources: ERC20BridgeSource[],
+        makerToken: string,
+        takerToken: string,
+        takerAmount: BigNumber,
+    ): Promise<Array<Array<DexSample<FillData>>>> {
+        const rpcLiquidityRequests: RpcLiquidityRequest[] = sources.map(source => {
+            return {
+                tokenPath: [makerToken, takerToken],
+                inputAmount: takerAmount.toString(),
+                source,
+                demand: true,
+            };
+        });
+        const rpcLiquidityResponse = await this.rpcSamplerClient.getSellLiquidityAsync(rpcLiquidityRequests);
+        const dexQuotes: Array<Array<DexSample<FillData>>> = rpcLiquidityResponse.map((response, i) => {
+            const dexSample: Array<DexSample<FillData>> = response.liquidityCurve.map((point, j) => {
+                const fillData: DexSample = {
+                    source: response.source,
+                    fillData: point.encodedFillData,
+                    input: point.sellAmount,
+                    output: point.buyAmount,
+                };
+                return fillData;
+            });
+            return dexSample;
+        });
+        return dexQuotes;
+    }
+
+    public async getBuyQuotesAsync(
+        sources: ERC20BridgeSource[],
+        makerToken: string,
+        takerToken: string,
+        takerAmount: BigNumber,
+    ): Promise<Array<Array<DexSample<FillData>>>> {
+        const rpcLiquidityRequests: RpcLiquidityRequest[] = sources.map(source => {
+            return {
+                tokenPath: [makerToken, takerToken],
+                inputAmount: takerAmount.toString(),
+                source,
+                demand: true,
+            };
+        });
+        const rpcLiquidityResponse = await this.rpcSamplerClient.getBuyLiquidityAsync(rpcLiquidityRequests);
+        const dexQuotes: Array<Array<DexSample<FillData>>> = rpcLiquidityResponse.map((response, i) => {
+            const dexSample: Array<DexSample<FillData>> = response.liquidityCurve.map((point, j) => {
+                const fillData: DexSample = {
+                    source: sources[i],
+                    fillData: point.encodedFillData,
+                    input: point.sellAmount,
+                    output: point.buyAmount,
```

Comment:
Flipped for buys. Not that buys work right now. I would just leave this whole function out.
```suggestion
                    input: point.buyAmount,
                    output: point.sellAmount,
```